#VRML_SIM R2022a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.06056747363412923 -0.6824051184806281 -0.7284605929005488 2.629091891591753
  position 1.1311964095402065 0.6661063331303906 0.07016157890882768
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation -0.5773502691896258 0.5773502691896258 0.5773502691896258 2.0943951023931957
  floorTileSize 1 1
  floorAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "C:/Users/ashut/Desktop/Paradoxial Bots/Arena_LineFollower.png"
      ]
      repeatS FALSE
      repeatT FALSE
      filtering 0
    }
    metalness 0
  }
}
Robot {
  translation 0.25483 0.0224255 0.154631
  rotation 0 1 0 1.5708
  children [
    DistanceSensor {
      translation -0.005 -0.0099999 -0.03
      rotation 0 0 1 -1.5707953071795862
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Box {
            size 0.003 0.003 0.003
          }
        }
      ]
      name "sensor_r"
      boundingObject USE sensor
      physics Physics {
      }
      type "infra-red"
      numberOfRays 10
      aperture 0.25
    }
    DistanceSensor {
      translation -0.005 -0.00999991 0.03
      rotation 0 0 -0.9999999999999999 1.5707953071795862
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Box {
            size 0.003 0.003 0.003
          }
        }
      ]
      name "sensor_l"
      boundingObject USE sensor
      physics Physics {
      }
      type "infra-red"
      numberOfRays 10
      aperture 0.25
    }
    DistanceSensor {
      translation -0.015 -0.00999992 0
      rotation 0 0 1 -1.5707953071795862
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Box {
            size 0.003 0.003 0.003
          }
        }
      ]
      name "sensor_m"
      boundingObject USE sensor
      physics Physics {
      }
      type "infra-red"
      numberOfRays 10
      aperture 0.25
    }
    Solid {
      translation -0.03 -0.0125 0
      children [
        DEF ball_f Shape {
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "solid(3)"
      boundingObject USE ball_f
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.01 0.00249999 0.04
      }
      device [
        RotationalMotor {
          name "motor_r"
        }
      ]
      endPoint Solid {
        translation 0.010000061573225832 0.0024996907977389484 0.04000000000000076
        rotation -1 2.63353999999468e-13 2.1637099999999696e-16 3.1415853071795863
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.01 0.0025 -0.04
      }
      device [
        RotationalMotor {
          name "motor_l"
        }
      ]
      endPoint Solid {
        translation 0.010000061573222668 0.0024997007977516136 -0.03999999999999926
        rotation 0.9999999999992921 -2.350349999998326e-10 -1.1897759197857347e-06 1.0150305250839343e-06
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    Transform {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.025 0.075
          }
        }
      ]
    }
  ]
  name "robo"
  boundingObject Box {
    size 0.1 0.025 0.075
  }
  physics Physics {
  }
  controller "control"
}
